protobuf/v1/replica.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "common.proto";
+
+package mayastor.v1;
+
+service ReplicaRpc {
+  // Replica related methods.
+  //
+  // Replica allocates space from storage pool.
+  rpc CreateReplica (CreateReplicaRequest) returns (Replica) {}
+  rpc DestroyReplica (DestroyReplicaRequest) returns (google.protobuf.Empty) {}
+  rpc ListReplicas (ListReplicasOptions) returns (ListReplicasResponse) {}
+  rpc ShareReplica (ShareReplicaRequest) returns (ShareReplicaResponse) {}
+}
+
+message CreateReplicaRequest {
+  string name = 1;  // name of the replica
+  string uuid = 2;  // uuid of the replica
+  string pool = 3;  // name of the pool
+  PoolType pooltype = 4; // type of the pool
+  uint64 size = 5;  // size of the replica in bytes
+  bool thin = 6;    // thin provisioning
+  ShareProtocol share = 7;  // protocol to expose the replica over
+}
+
+// Destroy replica arguments.
+message DestroyReplicaRequest {
+  string uuid = 1;  // uuid of the replica
+}
+
+// Replica properties
+message Replica {
+  string name = 1;  // name of the replica
+  string uuid = 2;  // uuid of the replica
+  string pool = 3;  // name of the pool
+  PoolType pooltype = 4; // type of the pool
+  bool thin = 5;    // thin provisioning
+  uint64 size = 6;  // size of the replica in bytes
+  ShareProtocol share = 7;  // protocol used for exposing the replica
+  string uri = 8;   // uri usable by nexus to access it
+}
+
+// List of replicas and their properties.
+message ListReplicasResponse {
+  repeated Replica replicas = 1;  // list of the replicas
+}
+
+// Share replica request.
+message ShareReplicaRequest {
+  string uuid = 1;  // uuid of the replica
+  // protocol used for exposing the replica
+  // Use "NONE" to disable remote access.
+  ShareProtocol share = 2;
+}
+
+// Share replica response.
+message ShareReplicaResponse {
+  string uri = 1;   // uri under which the replica is accessible by nexus
+}
+
+message ListReplicasOptions {
```

Comment:
What is the idea here? If you specify neither `name` nor `poolname`, you get all pools of `pooltype`?
Is `PoolTypeValue` an option?

Reply:
This is the generated code.
```rust
pub struct ListReplicasOptions {
    #[prost(message, optional, tag = "1")]
    pub name: ::core::option::Option<::prost::alloc::string::String>,
    #[prost(message, optional, tag = "2")]
    pub poolname: ::core::option::Option<::prost::alloc::string::String>,
    #[prost(message, optional, tag = "3")]
    pub pooltype: ::core::option::Option<PoolTypeValue>,
```

So if you dont specify anything, all fields will be none and all replicas will be listed. You can set some fields, for eg: `PoolType: Lvs`, will list all replicas on pools that are backed by lvolstore.  Similarly with name, and pool name.

Reply:
Cool thank you for explaining :+1: 
Should we have replica uuid as well?

Reply:
Thought that at first, we will have just the simple and basic stuff in the message so the current functionality works as is and this forms a basis for later more complex selections that we can add. But IMO, just based on name selection should be good for now , and it can always be extended.